scenarios/two_party_countersigning/src/main.rs

```diff
@@ -507,10 +614,20 @@ async fn await_countersigning_success(
                         break;
                     }
                     Signal::System(SystemSignal::SuccessfulCountersigning(_)) => {
-                        // This shouldn't happen because the scenario should only be running one at a time. There's a bug if this log message shows up.
+                        // This shouldn't happen because only one countersigning session can be active at a time. There's a bug if this log message shows up.
                         log::warn!("Received a successful countersigning signal for a different session, listening for other signals.");
                         continue;
                     }
+                    Signal::System(SystemSignal::AbandonedCountersigning(eh))
+                        if eh == session_entry_hash =>
+                    {
+                        return Err(anyhow::anyhow!("Countersigning session was abandoned"));
+                    }
+                    Signal::System(SystemSignal::AbandonedCountersigning(_)) => {
+                        // This shouldn't happen because only one countersigning session can be active at a time. There's a bug if this log message shows up.
+                        log::warn!("Received an abandoned countersigning signal for a different session, listening for other signals.");
+                        continue;
```

Comment:
Given the comment then should we log this as an error instead?

Reply:
Fair point, yes